pkg/query/columnquery.go

```diff
@@ -732,9 +733,35 @@ func ComputeDiff(ctx context.Context, tracer trace.Tracer, base, compare profile
 	_, span := tracer.Start(ctx, "ComputeDiff")
 	defer span.End()
 
+	type totals struct {
+		value          int64
+		valuePerSecond float64
+	}
+	totalsCompare := make([]totals, len(compare.Samples))
+	for i, s := range compare.Samples {
+		totalsCompare[i].value = math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalsCompare[i].valuePerSecond = math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
+	totalsBase := make([]totals, len(compare.Samples))
+	for i, s := range base.Samples {
+		totalsBase[i].value = math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalsBase[i].valuePerSecond = math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
+
 	records := make([]arrow.Record, 0, len(compare.Samples)+len(base.Samples))
+	for i, r := range compare.Samples {
+		var cumulativeRatio, cumulativePerSecondRatio float64
+		if totalsBase[i].value > totalsCompare[i].value {
```

Comment:
I don't understand these. We need to scale up or down based on the ratio, why would we only scale up if the base is larger?

Reply:
It's all about the percentage between the two profiles. Therefore, I opted to always scale up the side with the lower cumulative total value. It would avoid essentially dividing and instead always multiplying the smaller side. 
During development I've seen quite some profiles (granted it was single profiles) where values were simply scale to `0` because the ratio was something like `0.1` or less.

Reply:
So always upscaling makes sense, but there are several problems here:

doing this on a per-record basis is wrong, we need to just sum up all values/per-second-values for all records of base/compare
the else statements don't make sense, just have a single ratio for base and compare that everything is multiplied by (if you want to optimize not multiplying by 1 since it's a no-op then handle that at the place where we perform the multiplication)

The way the code is right now, even if maybe correct, I can't follow it at all.

Reply:
You're right. I'll refactor to sum up the total for all records of each base and compare.